sled-hardware/src/lib.rs

```diff
@@ -73,7 +73,7 @@ pub enum SledMode {
 }
 
 /// Describes properties that should uniquely identify a Gimlet.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
```

Comment:
I can't comment on line 91 because you didn't actually change it, but FYI, I'd have some caution depending on `Baseboard::unknown` -- that's kinda on my "gross hacks to clean up" list, rather than my "public API we should be really using" list.